metrics-operator/controllers/common/analysis/interfaces.go

```diff
@@ -0,0 +1,26 @@
+package analysis
+
+import (
+	"github.com/keptn/lifecycle-toolkit/metrics-operator/api/v1alpha3"
+	"github.com/keptn/lifecycle-toolkit/metrics-operator/controllers/common/analysis/types"
+)
+
+//go:generate moq -pkg fake -skip-ensure -out ./fake/analysis_evaluator_mock.go . IAnalysisEvaluator
+type IAnalysisEvaluator interface {
+	Evaluate(values map[string]string, ad v1alpha3.AnalysisDefinition) types.AnalysisResult
```

Comment:
```suggestion
	Evaluate(values map[string]string, ad *v1alpha3.AnalysisDefinition) types.AnalysisResult
```

Reply:
Do we have a reason why this needs to be a pointer ?

Reply:
I think we could potentially have many objectives passed here, no? So pass-by copy may be heavy?